src/Combinators/notGuard.ts

```diff
@@ -0,0 +1,32 @@
+import {ReasonGuard} from '../ReasonGuard';
+import {isNegatableGuard, NegatableGuard, buildNegatable} from '../NegatableGuard';
+
+export const notGuard =
+	<FROM, TO extends FROM, N extends FROM = FROM>
+	(inner: ReasonGuard<FROM, TO>|NegatableGuard<FROM, TO, N>) =>
+		isNegatableGuard(inner)
+			? inner.negate()
+			: buildNegatable(
+				() => getRawNot(inner),
+				() => inner
+			);
+
+function getRawNot<FROM, TO extends FROM>(
+	inner: ReasonGuard<FROM, TO>
+): ReasonGuard<FROM, FROM> {
+	return (input: FROM, errors: Error[] = [], confirmations: string[] = []): input is FROM => {
+		try {
+			const innerErrors: Error[] = [];
+			const innerConfs: string[] = [];
+			if (inner(input, innerErrors, innerConfs)) {
+				throw new Error(innerConfs[innerConfs.length-1]);
+			} else {
+				confirmations.push(innerErrors[0].message);
```

Comment:
Why not push all of the errors if it returns more than one?

Reply:
The problem is the relationship between multiple confirmations. Are they both necessary, or would either be good enough? The logic I settled on is "any error is a failure" (i.e. errors aggregate by `or` -- if you have error A *or* error B you have an error) and "confirmations are all required" (i.e. confirmations aggregate by `and`).

This gets tricky at a negation boundary, because if you have two errors (either of these is a failure) negated you *don't* have to two confirmations (both of these are required). So instead one gets arbitrarily masked.

Again, a more complex hierarchical structure could potentially express more information, but I don't have a complete answer there yet.